### PR TITLE
convert "for example" into runnable examples

### PR DESCRIPTION
**Pull request by rogpeppe**
It's nicer to have actual working runnable examples
than comments that are easy to get wrong or leave stale.

### CODE REVIEW

File: checker.go [hunk 1]
```diff
@@ -526,13 +449,13 @@ func (c *stringContainsChecker[T]) Args() []Arg {
...
 // slice contains the given element, by comparing for equality.
-func SliceContains[T comparable](container []T, elem T) Checker {
+func SliceContains[T any](container []T, elem T) Checker {
 	return SliceAny(container, F2(Equals[T], elem))
 }
 
 // MapContains returns a Checker that succeeds if the given
 // map contains the given value, by comparing for equality.
-func MapContains[K, V comparable](container map[K]V, elem V) Checker {
+func MapContains[K comparable, V any](container map[K]V, elem V) Checker {
```
Comment (mvdan) [inline checker.go:458]: worth moving this to a separate commit with a test? or is one of the runnable examples already covering this?

File: example_test.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+	// Output:
+}
+
+func Example_ErrorIs() {
+	runExampleTest(func(t testing.TB) {
+		_, err := os.Open("/non-existent-file")
+
+		qt.Assert(t, qt.ErrorIs(err, os.ErrNotExist))
+	})
+	// Output:
```
Comment (mvdan) [inline example_test.go:295]: can we add a `fmt.Println("PASS")` at the end of `runExampleTest`, just for the sake of being very clear that the test passed? it also avoids the awkwardness of `// Output:` followed by... nothing :)

File: example_test.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+	"github.com/go-quicktest/qt"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func Example_Comment() {
+	runExampleTest(func(t testing.TB) {
+		a := 42
+		qt.Assert(t, qt.Equals(a, 42), qt.Commentf("no answer to life, the universe, and everything"))
```
Comment (mvdan) [inline example_test.go:20]: would be nice if this example actually failed, as then you show how the output looks like with the comment. might be tricky with filenames and such though.

File: example_test.go [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+			a: 1234,
+		}
+		qt.Assert(t, customDeepEquals(got, &myStruct{
+			a: 1234,
+		}))
+	})
+	// Output:
+}
+
+func Example_ContentEquals() {
```
Comment (mvdan) [inline example_test.go:71]: do you want the names to contain underscores? AFAIK to have an example for `Foo`, you want `ExampleFoo`, not `Example_Foo`. The end of https://pkg.go.dev/testing#hdr-Examples explains the syntax, and I think vet checks it to some degree.

**Review by mvdan — APPROVED** [review]

File: example_test.go (hunk 3 above)
Comment (rogpeppe) [reply example_test.go:20]: Yeah, I don't see a way to do that with the output containing a stack trace and all.

File: example_test.go (hunk 4 above)
Comment (rogpeppe) [reply example_test.go:71]: Ah, thanks! I always get this wrong and it's hard to know that it's wrong until the PR lands and one eventually sees the result inside pkg.go.dev.

File: example_test.go (hunk 2 above)
Comment (rogpeppe) [reply example_test.go:295]: good idea. Done.

File: example_test.go (hunk 4 above)
Comment (rogpeppe) [reply example_test.go:71]: done

File: checker.go (hunk 1 above)
Comment (rogpeppe) [reply checker.go:458]: good idea. Reverted for now.
